release 15.1 does not build on DragonFly

### Zig Version

N/A

### Steps to Reproduce and Observed Behavior

Release 14.1 builds on DragonFly.  The new release 15.1 does not.

```
--- stage3/bin/zig ---
[100%] Building stage3
cd /construction/zig/zig-0.15.1 && /construction/zig/.build/zig2 build --prefix /construction/zig/.b
lib/std/net.zig:2296:40: error: type 'void' has no members
                const flags = posix.MSG.NOSIGNAL;
                              ~~~~~~~~~^~~~~~~~~
referenced by:
    init: lib/std/net.zig:2230:30
    writer: lib/std/net.zig:2316:21
    16 reference(s) hidden; use '-freference-trace=18' to see all references
```

Any ideas?


### Expected Behavior

All stages are expected to compile.

--- a/lib/std/c.zig
+++ b/lib/std/c.zig
@@ -5825,6 +5825,22 @@ pub const MSG = switch (native_os) {
         pub const WAITFORONE = 0x1000;
         pub const CMSG_CLOFORK = 0x2000;
     },
+    .dragonfly => struct {
+        pub const OOB = 0x0001;
+        pub const PEEK = 0x0002;
+        pub const DONTROUTE = 0x0004;
+        pub const EOR = 0x0008;
+        pub const TRUNC = 0x0010;
+        pub const CTRUNC = 0x0020;
+        pub const WAITALL = 0x0040;
+        pub const DONTWAIT = 0x0080;
+        pub const NOSIGNAL = 0x0400;
+        pub const SYNC = 0x0800;
+        pub const CMSG_CLOEXEC = 0x1000;
+        pub const CMSG_CLOFORK = 0x2000;
+        pub const FBLOCKING = 0x10000;
+        pub const FNONBLOCKING = 0x20000;
+    },
     else => void,
 };
 pub const SOCK = switch (native_os) {

